restock-shop.lic

```diff
@@ -224,50 +201,82 @@ class ShopRestock
       end
     end
 
-    while bput('get silk cloth from my back', 'You get', 'What were') == 'You get'
+    while bput("get silk cloth from my #{@bag}", 'You get', 'What were') == 'You get'
       fput('combine cloth')
     end
     stow_hands
 
     missing_items.each do |item|
       move('out') if Room.current.id == @inner_id
-      wait_for_script_to_complete('sew', ['stow', 'sewing', item[4], item[5], item[0]])
-      walk_to(@shop_id)
-      fput('open entrance')
-      pause 0.5
-      fput('go entrance')
-      fput("get #{item[0]} from back")
-      fput('get my stamp')
-      fput("mark my #{item[0]} with my stamp")
-      waitrt?
-      pause
-      fput('stow stamp')
-      fput("sell my #{item.first} #{item[3]} for #{item[1]}")
+      wait_for_script_to_complete('sew', ['stow', 'sewing', item['chapter'], item['recipe'], item['noun']])
+      go_to_shop
+      stamp(item)
+      sell(item)
     end
   end
 
+  def buy_part(name)
+    part_data = @part_data[name][@town]
+    if part_data['part-number']
+      order_item(part_data['part-room'], part_data['part-number'])
+    else
+      buy_item(part_data['part-room'], name)
+    end
+    stow_hands
+  end
+
   def restock_gryphon(missing_items)
     return if missing_items.empty?
-    echo missing_items
 
-    order_item(16_667, 16)
-    order_item(16_667, 17)
-    stow_hands
+    missing_items.each do |item|
+      recipe_data = recipe_lookup(@recipes, item['recipe'])
+      recipe_data['part'].each { |part| buy_part(part) }
+      wait_for_script_to_complete('sew', ['stow', item['material'], recipe_data['chapter'], item['recipe'], item['noun'], 'gryphon'])
+      fput('stow leather')
+      fput("get shield from my #{@bag}")
+      wait_for_script_to_complete('sew', ['seal'])
+      go_to_shop
+      stamp(item)
+      sell(item)
+    end
+  end
+
+  def go_to_shop
+    return if Room.current.id == @inner_id
 
-    wait_for_script_to_complete('sew', ['stow', 'leather', 10, 'small leather shield', 'shield', 'gryphon'])
-    fput('stow leather')
-    fput('get shield from my back')
-    wait_for_script_to_complete('sew', ['seal'])
     walk_to(@shop_id)
-    fput('open entrance')
+    if bput("open #{@entrance}",
+            'You will need to contact the owner about that',
+            'But your shop is already set up and ready for you to enter',
+            'Your shop is ready') == 'You will need to contact the owner about that'
+      echo '*****************'
+      echo 'Your shop settings appear to be incorrect. Please check the script documentation and update your settings'
+      echo '*****************'
+      exit
+    end
     pause 0.5
-    fput('go entrance')
+    move("go #{@entrance}")
+  end
+
+  def sell(item)
+    fput("sell my #{item['noun']} #{item['surface']} for #{item['price']}")
+  end
+
+  def stamp(item)
+    # TODO: Convert to stow/get crafting item
```

Comment:
❓ Is it worth waiting to merge this until this TODO is resolved?

Reply:
I have implemented this on my end, ill push the commit after some rounds of testing

Reply:
👍

Reply:
Removing stow_crafting_item as it just put my stamp in a backpack i was crafting rather than my own. get_crafting_item is working fine though.